tools/@aws-cdk/spec2cdk/lib/cdk/resource-decider.ts

```diff
@@ -179,7 +179,7 @@ export class ResourceDecider {
 
   private handleTagPropertyModern(cfnName: string, prop: Property, variant: TagVariant) {
     const originalName = propertyNameFromCloudFormation(cfnName);
-    const originalType = this.converter.typeFromProperty(prop);
+    const originalType = this.converter.typeFromPropertyForModernTags(prop);
```

Comment:
This should be safe because `handleTagPropertyModern` is only enabled in version 2.131.0 which is the current CDK version, see this PR https://github.com/aws/aws-cdk/pull/28989/files.

Before version 2.131.0 there would be no way to reach this function call due to `HAS_25610` is always set to `false` (see the PR I linked). And for 2.131.0 there wouldn't be a deployable template since it will fails at deployment step, see description.